service-api/src/main/java/greencity/dto/commitinfo/CommitInfoDto.java

```diff
@@ -0,0 +1,12 @@
+package greencity.dto.commitinfo;
+
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+/**
+ * Base class for commit information DTO.
+ */
+@Data
+@NoArgsConstructor
+public abstract class CommitInfoDto {
```

Comment:
can be an interface

Reply:
Or itis  better to create a single class with the next fields
```
 private String commitHash;
 private String commitDate;
 private String error;
```
And in the controller check if the error is present

Reply:
1) The interface requirement is satisfactory.
2) It would be advantageous to have two types of responses. This is elegantly implemented through polymorphism. It is somewhat unappealing to have null in the error field.